.github/workflows/test.yml

```diff
@@ -7,50 +7,213 @@ on:
       - master
       - develop
 
+concurrency:
+  group: axe-core Tests ${{ github.ref_name }}
+  cancel-in-progress: ${{ github.event_name != 'push' }}
+
+permissions: {}
+
+env:
+  CHROME_DEVEL_SANDBOX: /opt/google/chrome/chrome-sandbox
+
 jobs:
-  build:
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
+  lint:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
     steps:
-      - uses: actions/checkout@v5
-      - uses: actions/setup-node@v6
+      - &checkout
+        name: Checkout repository
+        uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0
+      - &setup-node
+        name: Set up Node.js
+        uses: actions/setup-node@2028fbc5c25fe9cf00d9f06a71cc4710d4507903 # v6.0.0
         with:
           node-version-file: .nvmrc
           cache: 'npm'
-      - run: npm ci
-      - run: npm run prepare
-      - run: npm run build
-      - uses: actions/upload-artifact@v5
+      - &install-deps-directly
+        name: Install Dependencies
+        run: npm ci
+      - name: Run ESLint
+        run: npm run eslint
+
+  fmt_check:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
+    steps:
+      - *checkout
+      - *setup-node
+      - *install-deps-directly
+      - run: npm run fmt:check
+
+  build:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
+    steps:
+      - *checkout
+      - *setup-node
+      - *install-deps-directly
+      - &build
+        name: Build
+        run: |
+          npm run prepare
+          npm run build
+      - uses: actions/upload-artifact@330a01c490aca151604b8cf639adc76d48f6c5d4 # v5.0.0
         with:
           name: axe-core
           path: axe.js
           retention-days: 1
 
-  fmt_check:
-    runs-on: ubuntu-latest
-    timeout-minutes: 5
+  test_chrome:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 10
+    env:
+      DISPLAY: :99
     steps:
-      - uses: actions/checkout@v5
-      - uses: actions/setup-node@v6
+      - *checkout
+      - &install-deps-with-xvfb
```

Comment:
So if I understand everything correctly, calling `&install-deps-with-xvfb` will reinstall the node modules for every step and not cache it, whereas in circle we found a way to cache the build and the node modules so it only happened once for the entire run. This isn't so much a problem for now as as you said for the build we can worry about optimizing after. Looking at the runs, each of the tests take ~2mins longer than their circleCI counterpart, but since it all runs in parallel it doesn't add too much time in the long run.